stylix/darwin/default.nix

```
@@ -20,4 +20,26 @@ in
     "${inputs.self}/stylix/target.nix"
     (import ../templates.nix inputs)
   ] ++ autoload;
+  config.warnings =
+    lib.mkIf
+      (
+        config.stylix.enable
+        && config.stylix.enableReleaseChecks
+        && (config.stylix.release != config.system.darwin.release)
+      )
+      [
+        ''
+          You are using different Stylix and nix-darwin versions. This is
+          likely to cause errors and unexpected behavior. It is highly
+          recommended that you use a version of Stylix that matches your chosen
+          version of nix-darwin.
+
+          If you are willing to accept the risks that come with using
+          mismatched versions, you may disable this error by adding
+
+            stylix.enableReleaseChecks = false;
```

Although Home Manager used used two spaces for indentation, should we use four spaces as it is more visible:

```suggestion
              stylix.enableReleaseChecks = false;
```